packages/core/src/components/DataTable/index.tsx

```diff
@@ -0,0 +1,421 @@
+      selectedRows,
+    }: { sortedDataList: IndexedParentRow[]; selectedRows: SelectedRows } = this.state;
+
+    const { parentOriginalIndex, parentIndex, originalIndex } = row.metadata;
+
+    if (parentOriginalIndex !== undefined && parentIndex !== undefined) {
+      if (Object.prototype.hasOwnProperty.call(selectedRows, parentOriginalIndex)) {
+        const { selectedChildren } = selectedRows[parentOriginalIndex];
+        if (selectedChildren.has(originalIndex)) {
+          selectedChildren.delete(originalIndex);
+          if (selectedChildren.size === 0) {
+            delete selectedRows[parentOriginalIndex];
+          } else {
+            selectedRows[parentOriginalIndex].status = SELECTION_OPTIONS.HAS_ACTIVE_CHILD;
+          }
+        } else {
+          selectedChildren.add(originalIndex);
+          if (sortedDataList[parentIndex].metadata.children.length === selectedChildren.size) {
+            selectedRows[parentOriginalIndex].status = SELECTION_OPTIONS.ACTIVE;
+          } else {
+            selectedRows[parentOriginalIndex].status = SELECTION_OPTIONS.HAS_ACTIVE_CHILD;
+          }
+        }
+      } else {
+        selectedRows[parentOriginalIndex] = {
+          status: SELECTION_OPTIONS.HAS_ACTIVE_CHILD,
+          selectedChildren: new Set([originalIndex]),
+        };
+      }
+    }
+
+    this.setState({
+      sortedDataList,
+    });
+  };
+
+  private handleParentSelection(row: ExpandedRow) {
+    const { selectedRows }: { selectedRows: SelectedRows } = this.state;
+    const { preExpandedIndex, originalIndex } = row.metadata;
+
+    if (typeof preExpandedIndex !== 'undefined') {
+      if (Object.prototype.hasOwnProperty.call(selectedRows, originalIndex)) {
+        delete selectedRows[originalIndex];
+      } else {
+        const children = row.metadata.children
+          ? row.metadata.children.map((child: IndexedChildRow) =>
+              child.metadata ? child.metadata.originalIndex : -1,
+            )
+          : [];
+        selectedRows[originalIndex] = {
+          status: SELECTION_OPTIONS.ACTIVE,
+          selectedChildren: row.metadata && row.metadata.children ? new Set(children) : new Set(),
+        };
+      }
+    }
+
+    this.setState({ selectedRows });
+  }
+
+  private handleSelection = (rowData: ExpandedRow) => () => {
+    if (rowData.metadata.isChild) {
+      this.handleChildSelection(rowData);
+    } else {
+      this.handleParentSelection(rowData);
+    }
+  };
+
+  // Have to use any to match react-virutalized's specified callback signature
+  private handleRowClick = ({ rowData }: { rowData: any }): void => {
+    this.props.selectOnRowClick && this.handleSelection(rowData)();
+  };
+
+  renderTableHeader(parentWidth: number) {
+    const {
+      editable,
+      instantEdit,
+      extraHeaderButtons,
+      rowHeight,
+      tableHeaderLabel,
+      tableHeaderHeight,
+    } = this.props;
+
+    const { editMode, selectedRows } = this.state;
+
+    return (
```

Comment:
can we update this to conditionally render? If we don't want a header it'd be nice to not have a tall empty node above the table. (could also do this in the `TableHeader` component)

```jsx
const shouldRender = editable || extraHeaderButtons.length > 0 ||  !! tableHeaderLabel;

return shouldRender ? <TableHeader /> : null;
```

Reply:
Haven't dug into find where it's rendered, but there also looks like there might be an empty row at the bottom of the table. can we also conditionally render that?

Reply:
Just to show what I mean, if I wrap the table with a drop shadow you can see it 
<img src="https://user-images.githubusercontent.com/4496521/57097196-fa98f580-6ccb-11e9-82cb-27b1429c7d64.png" width="300" />